### PR TITLE
Add a few commands for working with the data layout.

### PR DESCRIPTION
**Pull request by kennykb**
Describe what your Pull Request is implementing
-----------------------------------------------

In order to implement 'sizeof' and 'alignof' correctly, and to get the LLVM coroutine frame built correctly, I needed to add interfaces to a few of the LLVM API's for working with target machines.

Version number advanced to 3.10.0 for the new interfaces.

Describe the platform you are using
-----------------------------------

Ubuntu 17.10, LLVM 5.0 from system package manager, Tcl 8..7 built from core-8-branch

### CODE REVIEW

File: generic/llvmtcl.cpp [hunk 1]
```diff
@@ -866,6 +866,39 @@ GarbageCollectUnusedFunctionsInModuleCmd(
...
 

+static int
+MakeTargetMachineCmd(ClientData clientData,
+		     Tcl_Interp* interp,
+		     int objc,
+		     Tcl_Obj *const objv[])
+{
+    if (objc < 1 || objc > 2) {
+	Tcl_WrongNumArgs(interp, 1, objv, "?Target?");
```
Comment (dkfellows) [inline generic/llvmtcl.cpp:876]: Missing `return TCL_ERROR`

**Review by dkfellows — CHANGES_REQUESTED** [review]
One nasty bug left.

**Comment by dkfellows** [comment]
I'm hoping I can get this into the 3.9 release, which I want to get out in the next couple of days.

**Comment by dkfellows** [comment]
https://github.com/dkfellows/llvmtcl/tree/kbk/target-data has my fixes.

**Comment by kennykb** [comment]
Thanks, Donal, for tidying up after me.  Right now the coroutine changes simply have too many moving parts and I'm leaving a bit of a mess wherever I go.

**Pull request closed without merging**